### ISSUE TITLE
A question about the logic of the code

### ISSUE THREAD

**Issue by ronghuautt** [ChrisLee63/person_search#6]
I read the original paper and the original caffe version code and your pytorch version code.
The original paper says: then we will update the t-th column of the LUT by v t ← γv t + (1 − γ)x, where γ ∈ [0, 1], and then scale v t to have unit L2-norm. (subsection 3.2). 
It says after update the feature table, the feature vector should be L2-normalized. 
caffe version code is:
caffe_gpu_axpby(K, (Dtype)1. - momentum_, bottom_data + i * K,
                      momentum_, weight + label_value * K);
and your code is:
lookup_table[label] = (
                    momentum * lookup_table[label] + (1 - momentum) * features[indx]
                )
But where is the L2-normalization part and do you think this is normal? (Normally L2-Norm is very necessary to produce cosine similarity.)
I think the authors of the paper are not active at the project anymore, so I am here to discuss with you. Maybe you can give me some response. Thanks.

**Comment by ChrisLee63** [comment]
Official caffe code does not normalize the feature in the lookup table, so I also don't.
It seems a bug of the official implementation.
You may be interested in the [issue](https://github.com/ShuangLI59/person_search/issues/128) as well.
Although the features in the lookup table are not normalized, the final output features are [normalized](https://github.com/ChrisLee63/person_search/blob/65199eee8baa25f2b2dbd80dfd9bbd713df2b580/lib/models/network.py#L94).

**Comment by ronghuautt** [comment]
Yes, the final feature vector normalized. The training result of this project is pretty nice. I am just curious about the training process. As for the issue you mentioned, it is obviously another bug. Maybe CQ is not so important in his implementation. 1/4 information CQ almost contributes nothing to the final loss function. But we still have to admit the final result is good.

**Comment by ChrisLee63** [comment]
Feel free to reopen the issue if you have any questions.
